course3/homework-list/test.cc

```diff
@@ -0,0 +1,89 @@
+#include "list.h"
+
+#include <cassert>
+#include <iostream>
+
+
+void list_test()
+{
+	// default constructor
+	Karun::List<int> l;
+	assert(l.empty());
+	assert(l.size() == 0);
+	assert(l.begin() == l.end());
+	
+	// push_front
+	l.push_front(10);
+	assert(l.size() == 1);
+	assert(l.front() == 10);
+	l.push_front(20);
+	assert(l.size() == 2);
+	assert(l.front() == 20);
+	l.push_front(30);
+	assert(l.size() == 3);
+	assert(l.front() == 30);
+	
+	// pop_front
+	l.pop_front();
+	assert(l.size() == 2);
+	assert(l.front() == 20);
+	l.pop_front();
+	assert(l.size() == 1);
+	assert(l.front() == 10);
+	l.pop_front();
+	assert(l.empty());
+	assert(l.begin() == l.end());
+	
+	// front
+	l.push_front(50); l.push_front(40); l.push_front(3); l.push_front(20); l.push_front(1);
+	assert(l.size() == 5);
+	assert(l.front() == 1);
+	l.front() *= 10;
+	assert(l.front() == 10);
+	
+	// iterator
+	int i = 0;
+	for (Karun::List<int>::iterator it = l.begin(); it != l.end(); ++it)
+	{
+		++i;
+		if (*it == 3) *it *= 10;
+		assert(*it == i * 10);
+	}
+	assert (i == 5);
+	
+	// const_iterator
```

Comment:
This doesn't work, since l.begin() returns an iterator, and not a const_iterator. How do I ensure the const begin method gets called instead?